cmd/thanos/query.go

```diff
@@ -405,6 +421,7 @@ func runQuery(
 			level.Info(logger).Log("msg", "Listening for query and metrics", "address", httpBindAddr)
 			return errors.Wrap(http.Serve(l, mux), "serve query")
 		}, func(error) {
+			querierIsReady = false
```

Comment:
We need to make it concurrent-safe - each function in `g.Add`  runs in separate go routine and also your readiness handler access same variable concurrently

Reply:
also why not ready here? I think setting it to false in one `g.Add` closing is enough (and were it matters -> in gRPC listeninig I guess)